packages/lesswrong/lib/collections/votes/custom_fields.js

```diff
@@ -10,3 +14,95 @@ Votes.addField([
     }
   }
 ]);
+
+//
+// Do a query, with a custom loader for query batching. This effectively does a
+// find query, where all of the fields of the query are kept constant within a
+// query batch except for one field, which is converted from looking for a
+// specific value to being a {$in: [...]} query. The loader caches within one
+// http request, and is reset between http requests.
+//
+//   collection: The collection which contains the objects you're querying for
+//   loaderName: A key which identifies this loader. Calls to getWithLoader
+//     that share a loaderName will be batched together, and must have an
+//     identical baseQuery
+//   groupByField: The name of the field whose value varies between queries in
+//     the batch.
+//   id: The value of the field whose values vary between queries in the batch.
+//
+async function getWithLoader(collection, loaderName, baseQuery, groupByField, id)
+{
+  if (!collection.extraLoaders) {
+    collection.extraLoaders = {};
+  }
+  if (!collection.extraLoaders[loaderName]) {
+    collection.extraLoaders[loaderName] = new DataLoader(async docIDs => {
+      let query = baseQuery ? _.clone(baseQuery) : {};
```

Comment:
You should set the default for the baseQuery in the construction, and then you can remove this line. I.e. 

`(collection, loaderName, baseQuery = {}, groupByField, id)`